lib/chunky_png/chunk.rb

```diff
@@ -319,6 +319,45 @@ def content
       end
     end
 
+    # The Physical (pHYs) chunk specifies the intended pixel size or aspect
+    # ratio for display of the image.
+    #
+    # http://www.libpng.org/pub/png/spec/1.2/PNG-Chunks.html#C.pHYs
+    #
+    class Physical < Generic
+      attr_accessor :ppux, :ppuy, :unit
+
+      def initialize(ppux, ppuy, unit=:unknown)
+        raise ArgumentError, 'unit must be either :meters or :unknown' unless [:meters, :unknown].member?(unit)
+        super('pHYs')
+        @ppux, @ppuy, @unit = ppux, ppuy, unit
+      end
+
+      def dpix
+        ppux * INCHES_PER_METER if unit == :meters
+      end
+
+      def dpiy
+        ppuy * INCHES_PER_METER if unit == :meters
```

Comment:
Well, good question!

Here's a bit of the spec:

> The following values are defined for the unit specifier:
> 
>   0: unit is unknown
>   1: unit is the meter
>
> When the unit specifier is 0, the pHYs chunk defines pixel aspect ratio only; the actual size of the pixels remains unspecified.

That made me think that raising an exception isn't the right move — like if the unit is `0`/`:unknown`, it's not an exceptional situation; and there's even still value in knowing the aspect ratio (`ppux / ppuy`).

Of course, `nil`'s not an informative return value. What do you think of these two methods returning `:unknown` instead of `nil`?

Or do you think they should just raise an exception and calling code should check the value of `unit` before attempting to call `dpix` and `dpiy`?